lib/library.rb

```diff
@@ -0,0 +1,59 @@
+require 'yaml'
+
+class Library
+  attr_accessor :books
+
+  def initialize
+    @books = load_item_info
+  end
+
+  def load_item_info
+     YAML.load_file('./lib/books.yml')
+  end
+
+  def print_item_info(item)
+    item[:title] + " " + "by" + " " + item[:author]
+  end
+
+  def available_book_list
+    listed_books = available_books
+    listed_books.each do |book_info|
+      puts "*******************************************"
```

Comment:
We're hardly going to build applications that output to the console (terminal) .. so quite often when you're implementing methods, make sure they return something that can be used by the caller. The caller will then be responsible for rendering/displaying in any way necessary.

In sum, stay away from `puts` as much as you can. Could be useful for debugging purposes though.